src/Maestro/Maestro.DataProviders/DarcRemoteFactory.cs

```diff
@@ -43,50 +43,63 @@ public async Task<IRemote> GetRemoteAsync(string repoUrl, ILogger logger)
     {
         using (_operations.BeginOperation($"Getting remote for repo {repoUrl}."))
         {
-            // Normalize the url with the AzDO client prior to attempting to
-            // get a token. When we do coherency updates we build a repo graph and
-            // may end up traversing links to classic azdo uris.
-            string normalizedUrl = AzureDevOpsClient.NormalizeUrl(repoUrl);
+            IRemoteGitRepo remoteGitClient = await GetRemoteGitClient(repoUrl, logger);
+            return new Remote(remoteGitClient, _versionDetailsParser, logger);
+        }
+    }
 
-            long installationId = await _context.GetInstallationId(normalizedUrl);
-            var repoType = GitRepoUrlParser.ParseTypeFromUri(normalizedUrl);
+    public async Task<IDependencyFileManager> GetDependencyFileManagerAsync(string repoUrl, ILogger logger)
```

Comment:
For future consideration: the `GetDependencyFileManagerAsync` method in both files is identical, and the logic behind `GetRemoteGitClient` is very similar. One day, we may consider having a base class ..
I am not sure if it's possible to cover the solution with unit tests, but it would be great to at least create a follow-up issue for that. We have interfaces everywhere so that it's possible to write tests.

Reply:
Fair point. I will be adding unit tests tomorrow so I will try to extract a common base class too. I think that should work.

I only prepared this very fast to squeeze it into the rollout so it's a bit messy.